### DISCUSSION TITLE
how to use join another table without relation

### DISCUSSION THREAD

**Discussion by lala7573** [mikro-orm/mikro-orm#2634] [Q&A] [answered]
Hi, folks.
I added a statistical table and I'd like to join with the original table without relation mapping on entities.
I have some sub queries, so I prefer to use QueryBuilder.
I've looked at the documentation, but it doesn't seem possible to join the table directly.
Is it possible to join a table without relation using QueryBuilder? Should I have to use raw query instead of it?
Any ideas would be helpful. 🙏

**Comment by B4nan** [comment, marked as answer]
The ORM query builder requires relations. So either use raw query, or knex. You can also use the ORM QB to create part of your query and then convert it to knex QB and join the additional stuff, there is `qb.getKnexQuery()` for that.

**Reply by jtomaszewski** [reply]
Would you be open to reconsidering this? One currently can use `raw()` in `.select()` or `.where()` functions.

Why don't we let doing the same in the existing join methods? (or at least, provide a new one, such as `.rawJoin(...)`)

**Reply by B4nan** [reply]
So what exactly would you like to do? I plan to get rid of knex in v7 which will also mean native support for the raw fragments pretty much everywhere.

Note that you can already join a subquery, this question is quite old (and very vague), it's not really clear what exactly you are missing.

**Reply by jtomaszewski** [reply]
I wanted to craft an inner join with a `VALUES LIST` ( https://www.postgresql.org/docs/current/queries-values.html ) using `raw()`. I know I can do `where: { [raw(...)]: ... }` but such query plan is less performant in postgres. Thus being able to do it with a JOIN would be better.

I'm sure there could be other usages for custom joins as well though.

> Note that you can already join a subquery,

A subquery which is still a query builder's query, right?

**Reply by B4nan** [reply]
> A subquery which is still a query builder's query, right?

ORM QB or a knex QB, in v7 the raw fragment will be also supported (I already have this part prepared locally).

Technically, knex.raw should work too, but wont probably type check now.

If you can show me a code you would like to work, I can give it a try and make it work, it should be relatively simple even in v6 I think.

**Reply by schulia** [reply]
Hey, I wanted to add to this just to give a clear picture of the need of allowing join with non related tables. I'm planning to integrate mikroorm to a legacy system in order to migrate dbs, so the plan is to not change the data but wire everything to orm so that we could move slowly and have more control over making it better. I really liked the sync between entities and db on mikroorm and plan to use it, but there is a lot of broken data in the system and any change is painful, so adding these relations would make our lives harder. I understand that we can use the querybuilder for it - and that's what I plan to do - but just giving you more context on why it could be nice to have it as an option.

**Comment by lala7573** [comment]
Thanks! I use knex to use join like below code :) 

```
const query = this.orm.em.createQueryBuilder(...)
const knex = this.orm.em.getKnex();
knex
            .select('agg.count', 'agg.id', 'name')
            .from(query.getKnexQuery().as('agg'))
            .leftJoin(knex('meta').as('m'), 'agg.id', 'm.id'),
```
